e2e/src/definition.spec.ts

```diff
@@ -0,0 +1,77 @@
+import assert = require('assert');
+import { Position, Range } from 'vscode';
+import { activateAndWait, getDocUri, triggerDefinition } from './helper';
+
+const refSqlDocUri = getDocUri('ref_sql.sql');
+const packageRefDocUri = getDocUri('package_ref.sql');
+
+suite('ref definitions', () => {
+  test('Should suggest definitions for ref without package', async () => {
+    // arrange
+    await activateAndWait(refSqlDocUri);
+
+    // act
+    const definitions = await triggerDefinition(refSqlDocUri, new Position(1, 24));
+
+    // assert
+    assert.strictEqual(definitions.length, 1);
+    assert.ok(definitions[0].targetUri.path.endsWith('/test-fixture/models/table_exists.sql'));
+    assert.deepStrictEqual(definitions[0].originSelectionRange, new Range(new Position(1, 19), new Position(1, 31)));
+  });
+
+  test('Should suggest definitions for ref with package', async () => {
+    // arrange
+    await activateAndWait(packageRefDocUri);
+
+    // act
+    const packageDefinitions = await triggerDefinition(packageRefDocUri, new Position(5, 24));
+    const modelDefinitions = await triggerDefinition(packageRefDocUri, new Position(5, 42));
+
+    // assert
+    assert.ok(packageDefinitions.length > 1);
+    assert.deepStrictEqual(packageDefinitions[0].originSelectionRange, new Range(new Position(5, 19), new Position(5, 33)));
+
+    assert.strictEqual(modelDefinitions.length, 1);
+    assert.ok(modelDefinitions[0].targetUri.path.endsWith('/test-fixture/models/table_exists.sql'));
+    assert.deepStrictEqual(modelDefinitions[0].originSelectionRange, new Range(new Position(5, 37), new Position(5, 49)));
+  });
+});
+
+suite('macro definitions', () => {
+  test('Should suggest definitions for macros', async () => {
+    // arrange
+    await activateAndWait(packageRefDocUri);
+
+    // act
+    const extractFirstNameDefinitions = await triggerDefinition(packageRefDocUri, new Position(2, 9));
+    const extractLastNameDefinitions = await triggerDefinition(packageRefDocUri, new Position(3, 9));
+
+    // assert
+    assert.strictEqual(extractFirstNameDefinitions.length, 1);
```

Comment:
We should create some function like `assertDefinitions(definitions: DefinitionLink[], expectedSize, originSelectionRange, targetSelectionRange,...)`, and check everything in this function, then call it from tests.

Reply:
Done where it is possible.